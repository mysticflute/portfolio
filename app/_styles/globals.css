/* The file contains root-level custom properties and element resets only. */

:root {
  /* colors */
  --color-gray-100: #0b0b0b;
  --color-gray-200: #393939;
  --color-gray-300: #5e5e5f;
  --color-gray-400: #878787;
  --color-neutral-100: #fff;
  --color-neutral-200: #eff0f6;
  --color-neutral-300: #eceae8;
  --color-red-100: #e85d75;
  /* --color-red-100: #e574bc; */
  --color-yellow-100: #f6b468;
  --color-blue-100: #53abf8;
  --color-purple-100: #4331f0;
  --color-purple-200: #5358f8;
  --color-primary: var(--color-purple-100);
  --color-accent-100: var(--color-purple-200);
  --color-accent-200: var(--color-blue-100);
  --color-accent-300: var(--color-yellow-100);
  --color-accent-400: var(--color-red-100);
  --color-background-100: var(--color-gray-100);
  --color-background-200: var(--color-neutral-100);
  --color-background-300: var(--color-gray-100);
  --color-background-highlight: var(--color-blue-100);
  --color-text: var(--color-gray-200);
  --color-text-accent: var(--color-gray-100);
  --color-text-alt: var(--color-gray-100);
  --color-text-link: var(--color-gray-100);
  --color-text-highlight: var(--color-blue-100);
  --color-text-inverse: var(--color-neutral-200);

  /* spacing */
  --layout-max-width: 1300px;
  --layout-padding-x: 24px;
  --layout-padding-y: 0;
  --layout-padding: var(--layout-padding-y) var(--layout-padding-x);
  --layout-margin-x: auto;
  --layout-margin-y: 0;
  --layout-margin: var(--layout-margin-y) var(--layout-margin-x);
}

*,
::before,
::after {
  box-sizing: border-box;
}

html {
  scroll-behavior: smooth;
  scroll-padding: 2rem;
}

body {
  color: var(--color-text);
  background: var(--color-primary);
  font-family: var(--font-default), system-ui, sans-serif;
  font-weight: 500;
  font-size: 1.125rem;
  line-height: 1.667em;
}

@media screen and (max-width: 767px) {
  body {
    font-size: 1rem;
  }
}

h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  font-size: 1rem;
}

ol,
ul {
  margin: 0;
  padding: 0;
  list-style: none;
}

p {
  margin: 0;
  padding: 0;
}

audio,
canvas,
iframe,
img,
svg,
video {
  vertical-align: middle;
}

img {
  max-width: 100%;
  height: auto;
}

a {
  color: inherit;
  text-decoration: none;
}

@media screen and (max-width: 991px) {
  :root {
    --layout-max-width: calc(600px + var(--layout-padding-x) * 2);
  }
}

@media screen and (max-width: 767px) {
  :root {
    --layout-max-width: calc(500px + var(--layout-padding-x) * 2);
  }
}

@media screen and (max-width: 479px) {
  :root {
    --layout-padding-x: 20px;
  }
}
