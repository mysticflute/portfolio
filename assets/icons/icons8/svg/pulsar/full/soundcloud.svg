<?xml version="1.0" encoding="iso-8859-1"?>
<!-- Generator: Adobe Illustrator 23.0.3, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 48 48" style="enable-background:new 0 0 48 48;" xml:space="preserve">
<path style="fill:#FFAB91;" d="M27.796,35.917l14.495,0c2.812,0,5.328-2.02,5.667-4.811c0.406-3.345-2.194-6.189-5.458-6.189
	c-0.655,0-1.28,0.12-1.863,0.33c-0.302-4.282-3.862-7.664-8.22-7.664c-2.039,0-4.25,0.914-5.779,2.186
	C26.229,20.108,26,20.616,26,21.148v12.973C26,35.113,26.804,35.917,27.796,35.917z"/>
<g>
	<path style="fill:#18193F;" d="M2.884,34.717c0.577,0.385,1.25,0.769,1.923,0.961V22.796c-0.673,0.192-1.346,0.481-1.923,0.961
		V34.717z"/>
	<path style="fill:#18193F;" d="M13.459,17.22c-0.673,0.288-1.346,0.577-1.923,0.961v17.785h1.923V17.22z"/>
	<path style="fill:#18193F;" d="M7.691,35.966V22.604c-0.288-0.096-0.673-0.096-0.961-0.096c-0.288,0-0.673,0-0.961,0.096V35.87
		c0.288,0.096,0.673,0.096,0.961,0.096H7.691z"/>
	<path style="fill:#18193F;" d="M1.923,24.526C0.769,25.776,0,27.41,0,29.237c0,1.827,0.769,3.461,1.923,4.711V24.526z"/>
	<path style="fill:#18193F;" d="M19.227,17.22c-0.577-0.192-1.25-0.385-1.923-0.481v19.227h1.923V17.22z"/>
	<path style="fill:#18193F;" d="M8.652,21.45v14.516h1.923V18.95C9.806,19.623,9.133,20.489,8.652,21.45z"/>
	<path style="fill:#18193F;" d="M16.343,16.739c-0.673,0-1.346,0.096-1.923,0.192v19.035h1.923V16.739z"/>
</g>
<path style="fill:none;stroke:#18193F;stroke-width:3;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:10;" d="
	M21.421,34.399H40.23c3.463,0,6.27-2.807,6.27-6.27s-2.807-6.27-6.27-6.27c-0.747,0-1.46,0.137-2.124,0.377
	c-0.344-4.881-4.402-8.736-9.371-8.736c-2.672,0-5.084,1.114-6.796,2.904"/>
</svg>