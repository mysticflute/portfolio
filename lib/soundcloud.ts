const baseSoundCloudURL = 'https://w.soundcloud.com/player/';

/**
 * Params for building a SoundCloud URL.
 */
export type SoundCloudParams = {
  /**
   * Use a black background with white text (default false).
   */
  inverse?: boolean;

  /**
   * Autoplay the track after loading (default false).
   */
  auto_play?: boolean;

  /**
   * Show the username in the track title (default false).
   */
  show_user?: boolean;

  /**
   * Use this color (in css hex format) for the play button.
   */
  color?: string;
};

/**
 * Gets the URL to load a SoundCloud track.
 *
 * @param id The 10-digit SoundCloud track ID.
 * @param params Additional SoundCloud parameters.
 * @returns The URL.
 */
export function getUrlForTrack(id: number, params: SoundCloudParams) {
  const allParams: SoundCloudParams = {
    inverse: false,
    auto_play: false,
    show_user: false,
    color: '#ff5500',
    ...params,
  };

  let url = `${baseSoundCloudURL}/?url=https%3A//api.soundcloud.com/tracks/${id}`;

  url += Object.keys(allParams).reduce((accumulator, key) => {
    const value = allParams[key as keyof SoundCloudParams];
    if (value === undefined || value == null) {
      return accumulator;
    }
    return accumulator + `&${key}=${encodeURIComponent(value)}`;
  }, '');

  return url;
}
