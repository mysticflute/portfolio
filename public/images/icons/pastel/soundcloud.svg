<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="36pt" height="36pt" viewBox="0 0 36 36" version="1.1">
<g id="surface6301">
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(100%,100%,100%);fill-opacity:1;" d="M 29.476562 18.617188 L 28.433594 18.617188 C 28.492188 18.28125 28.519531 17.972656 28.519531 17.632812 C 28.519531 14.34375 25.929688 11.671875 22.753906 11.671875 C 21.039062 11.671875 19.492188 12.460938 18.421875 13.726562 L 18.421875 26.578125 L 29.476562 26.578125 C 31.585938 26.578125 33.328125 24.804688 33.328125 22.613281 C 33.328125 20.417969 31.613281 18.617188 29.476562 18.617188 Z M 29.476562 18.617188 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(100%,85.882354%,67.058825%);fill-opacity:1;" d="M 27.507812 21.9375 L 26.464844 21.9375 C 26.523438 21.601562 26.550781 21.289062 26.550781 20.953125 C 26.550781 17.664062 23.960938 14.992188 20.785156 14.992188 C 19.941406 14.992188 19.152344 15.1875 18.449219 15.523438 L 18.449219 26.578125 L 29.503906 26.578125 C 30.179688 26.578125 30.796875 26.410156 31.359375 26.101562 C 31.359375 26.042969 31.359375 25.988281 31.359375 25.929688 C 31.332031 23.710938 29.617188 21.9375 27.507812 21.9375 Z M 27.507812 21.9375 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(26.666668%,29.411766%,32.941177%);fill-opacity:1;" d="M 29.476562 27.421875 L 18.449219 27.421875 C 17.972656 27.421875 17.605469 27.054688 17.605469 26.578125 C 17.605469 26.101562 17.972656 25.734375 18.449219 25.734375 L 29.503906 25.734375 C 31.164062 25.734375 32.511719 24.328125 32.511719 22.613281 C 32.511719 20.898438 31.164062 19.492188 29.503906 19.492188 L 28.460938 19.492188 C 28.210938 19.492188 27.984375 19.378906 27.816406 19.179688 C 27.648438 18.984375 27.589844 18.730469 27.617188 18.507812 C 27.675781 18.195312 27.703125 17.914062 27.703125 17.664062 C 27.703125 14.851562 25.507812 12.542969 22.78125 12.542969 C 21.375 12.542969 20.054688 13.191406 19.097656 14.289062 C 18.789062 14.652344 18.253906 14.679688 17.914062 14.371094 C 17.550781 14.0625 17.523438 13.527344 17.832031 13.191406 C 19.097656 11.699219 20.898438 10.855469 22.808594 10.855469 C 26.464844 10.855469 29.417969 13.894531 29.417969 17.664062 C 29.417969 17.71875 29.417969 17.773438 29.417969 17.804688 L 29.53125 17.804688 C 32.117188 17.804688 34.226562 19.96875 34.226562 22.613281 C 34.171875 25.257812 32.0625 27.421875 29.476562 27.421875 Z M 29.476562 27.421875 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(26.666668%,29.411766%,32.941177%);fill-opacity:1;" d="M 15.132812 27.421875 C 14.652344 27.421875 14.289062 27.054688 14.289062 26.578125 L 14.289062 14.710938 C 14.289062 14.230469 14.652344 13.867188 15.132812 13.867188 C 15.609375 13.867188 15.976562 14.230469 15.976562 14.710938 L 15.976562 26.578125 C 15.976562 27.027344 15.609375 27.421875 15.132812 27.421875 Z M 15.132812 27.421875 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(26.666668%,29.411766%,32.941177%);fill-opacity:1;" d="M 11.8125 27.421875 C 11.335938 27.421875 10.96875 27.054688 10.96875 26.578125 L 10.96875 15.382812 C 10.96875 14.90625 11.335938 14.539062 11.8125 14.539062 C 12.289062 14.539062 12.65625 14.90625 12.65625 15.382812 L 12.65625 26.578125 C 12.65625 27.027344 12.289062 27.421875 11.8125 27.421875 Z M 11.8125 27.421875 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(26.666668%,29.411766%,32.941177%);fill-opacity:1;" d="M 8.523438 27.421875 C 8.042969 27.421875 7.679688 27.054688 7.679688 26.578125 L 7.679688 18.703125 C 7.679688 18.226562 8.042969 17.859375 8.523438 17.859375 C 9 17.859375 9.367188 18.226562 9.367188 18.703125 L 9.367188 26.578125 C 9.367188 27.027344 8.972656 27.421875 8.523438 27.421875 Z M 8.523438 27.421875 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(26.666668%,29.411766%,32.941177%);fill-opacity:1;" d="M 5.203125 24.101562 C 4.726562 24.101562 4.359375 23.738281 4.359375 23.257812 L 4.359375 21.992188 C 4.359375 21.515625 4.726562 21.148438 5.203125 21.148438 C 5.679688 21.148438 6.046875 21.515625 6.046875 21.992188 L 6.046875 23.257812 C 6.046875 23.738281 5.679688 24.101562 5.203125 24.101562 Z M 5.203125 24.101562 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(26.666668%,29.411766%,32.941177%);fill-opacity:1;" d="M 5.203125 27.421875 C 4.976562 27.421875 4.753906 27.335938 4.613281 27.167969 C 4.445312 27 4.359375 26.804688 4.359375 26.578125 C 4.359375 26.351562 4.445312 26.128906 4.613281 25.988281 C 4.640625 25.960938 4.695312 25.902344 4.753906 25.875 C 4.808594 25.847656 4.835938 25.820312 4.894531 25.789062 C 4.949219 25.761719 5.007812 25.761719 5.0625 25.734375 C 5.34375 25.679688 5.625 25.761719 5.820312 25.960938 C 5.851562 25.988281 5.90625 26.042969 5.933594 26.101562 C 5.960938 26.15625 5.992188 26.183594 6.019531 26.242188 C 6.046875 26.296875 6.046875 26.351562 6.074219 26.410156 C 6.074219 26.464844 6.101562 26.523438 6.101562 26.578125 C 6.101562 26.804688 6.019531 27.027344 5.851562 27.167969 C 5.652344 27.308594 5.429688 27.421875 5.203125 27.421875 Z M 5.203125 27.421875 "/>
</g>
</svg>
