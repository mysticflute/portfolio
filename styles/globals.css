:root {
  --color-dark: #0b0b0b;
  --color-blue: #1c92ff;
  --color-red: #ff4a60;
  --color-yellow: #ffc431;
  --color-text-default: #393939;
  --color-text-dark: var(--color-dark);
  --color-text-header-link: var(--color-dark);
  --color-text-header-link-highlight: var(--color-blue);
  --color-background-default: #faf9f6;
  --color-background-dark: var(--color-dark);
  --color-background-highlight: var(--color-blue);
}

*,
::before,
::after {
  box-sizing: border-box;
}

body {
  color: var(--color-text-default);
  background: var(--color-background-default);
  font-family: var(--font-default), system-ui, sans-serif;
  font-weight: 500;
  font-size: 1.125rem;
  line-height: 1.667em;
}

@media screen and (max-width: 767px) {
  body {
    font-size: 1rem;
  }
}

h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  font-size: 1rem;
}

ol,
ul {
  margin: 0;
  padding: 0;
  list-style: none;
}

p {
  margin: 0;
  padding: 0;
}

audio,
canvas,
iframe,
img,
svg,
video {
  vertical-align: middle;
}

img {
  max-width: 100%;
  height: auto;
}

a {
  color: inherit;
  text-decoration: none;
}

/* @media (prefers-color-scheme: dark) {
  html {
    color-scheme: dark;
  }
} */

/* @media (prefers-color-scheme: dark) {
  :root {
  }
} */
