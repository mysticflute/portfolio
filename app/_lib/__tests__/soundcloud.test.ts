import { describe, it, expect } from '@jest/globals';
import { getUrlForTrack } from '../soundcloud';

describe('soundcloud', () => {
  describe('getUrlForTrack', () => {
    it('should return the correct URL with no options specified', () => {
      expect(getUrlForTrack(1661989830)).toMatchSnapshot();
    });

    it('should return the correct URL with some options specified', () => {
      expect(
        getUrlForTrack(1661989830, {
          color: '#fffeee',
        }),
      ).toMatchSnapshot();
    });

    it('should return the correct URL with all options specified', () => {
      expect(
        getUrlForTrack(1661989830, {
          inverse: true,
          auto_play: true,
          show_user: true,
          color: '#fffeee',
        }),
      ).toMatchSnapshot();
    });
  });
});
