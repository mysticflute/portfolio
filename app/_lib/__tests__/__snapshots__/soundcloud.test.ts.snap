// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`soundcloud getUrlForTrack should return the correct URL with all options specified 1`] = `"https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/1661989830&inverse=true&auto_play=true&show_user=true&color=%23fffeee"`;

exports[`soundcloud getUrlForTrack should return the correct URL with no options specified 1`] = `"https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/1661989830&inverse=false&auto_play=false&show_user=false&color=%23ff5500"`;

exports[`soundcloud getUrlForTrack should return the correct URL with some options specified 1`] = `"https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/1661989830&inverse=false&auto_play=false&show_user=false&color=%23fffeee"`;
